### ISSUE TITLE
Table.removeColumns() by name or index removes all columns that contain identical data?

### ISSUE THREAD

**Issue by jglogan** [jtablesaw/tablesaw#689]
The following unit fails, with the actual column count being 0 instead of 2:

```
    @Test
    void testColumnDelete() throws IOException {
        final Table table = Table.create(BooleanColumn.create("a"), BooleanColumn.create("b"), BooleanColumn.create("c"));
        System.out.println(table.columnCount() + " " + table.columnNames().stream().collect(Collectors.joining(",")));
        assertEquals(3, table.columnCount());
        table.removeColumns(table.columnNames().indexOf("b"));
        System.out.println(table.columnCount() + " " + table.columnNames().stream().collect(Collectors.joining(",")));
        assertEquals(2, table.columnCount());
    }
```

Is this expected?

The cause is that Relation.removeColumns() function uses ArrayList.removeAll(column), which in turn relies on the override [BooleanColumn.equals()](https://github.com/jtablesaw/tablesaw/blob/master/core/src/main/java/tech/tablesaw/api/BooleanColumn.java#L668) to determine which columns to remove.  The equals() method only compares data, so all columns with matching data are removed.

**Comment by benmccann** [comment]
That definitely sounds like a bug. Thanks for debugging it!

If you want to take it a step further a PR would be welcomed :-)

**Comment by jglogan** [comment]
Thanks for confirming, Ben.  I'll look into a getting PR submitted.

**Comment by lwhite1** [comment]
turns out this only affected boolean columns. 

fixed in PR #766
